src/promptflow-azure/promptflow/azure/_utils/_token_cache.py

```diff
@@ -0,0 +1,37 @@
+import time
+
+from promptflow.core._connection_provider._utils import get_arm_token
+
+
+class SingletonMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = instance
+        return cls._instances[cls]
+
+
+class ArmTokenCache(metaclass=SingletonMeta):
+    DEFAULT_TTL_SECS = 1800
+
+    def __init__(self):
+        self._ttl_secs = self.DEFAULT_TTL_SECS
+        self._cache = {}
+
+    def _is_token_valid(self, entry):
+        return time.time() < entry["expires_at"]
```

Comment:
should we do refresh for token which is about to expire in near time?

azure-identity will do it for 300 seconds
https://github.com/Azure/azure-sdk-for-python/blob/9fb4c52d988ceb3b9583b2dff48d2e0304228590/sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py#L240

Reply:
I've updated the code to align with azure-identity's behavior.